### ISSUE TITLE
V9 Modular Authentication - Auth info is lost after closing the chrome extension

### ISSUE THREAD

**Issue by amitsingh-007** [firebase/firebase-js-sdk#5264]
### [REQUIRED] Describe your environment

  * Operating System version: Windows 10
  * Browser version: Google Chrome Version 92
  * Firebase SDK version: firebase@9.0.0-beta.8
  * Firebase Product: auth

### [REQUIRED] Describe the problem
Used the new modular firebase SDK in my chrome extension. Everything works except the authentication persistence. Everything works until the chrome extension popup is closed. Once closed, again if I open it and let's say update the firebase DB, it gives permission denied.
I have tried using persistence as `indexedDBLocalPersistence, browserLocalPersistence and browserSessionPersistence` but none of them seems to retain the authentication state. Although I can see some firebase authentication values stored in the local storage and indexedDB.
#### Steps to reproduce:
1. Open the chrome extension and log in using the below code.
2. Update the firebase DB. //this works, no permission error
3. Close the chrome extension.
4. Open the extension again and try to update the firebase DB. //gives permission error
#### Relevant Code:

```js
export const googleSignIn = async (token: string) => {
  const auth = initializeAuth(firebaseApp, {
    persistence: indexedDBLocalPersistence,
  });
  return await signInWithCredential(
    auth,
    GoogleAuthProvider.credential(null, token)
  );
};
```
#### Other things that I have tried
Currently, my logic resides in the chrome extension popup for authentication and DB updates. I have tried to move the authentication and firebase DB updates to the service-worker(in Manifest V3 or background scripts in Manifest V2) since it is never closed. It works as expected but as soon as I close the browser and reopen, the authentication state is lost and it again gives the permission denied error.

**Comment by sam-gc** [comment]
Thanks for spinning this off as its own issue. I've verified that indexed DB works in Manifest V2 with a background page and `signInWithPopup()`. When you try to access your database, do you wait until the auth state has initialized (i.e. by waiting until the [`auth.onAuthStateChanged()`](https://firebase.google.com/docs/reference/js/v9/auth.auth#authonauthstatechanged) callback fires)?

If so, can you inspect the chrome extension's dev tools and look to see if the user is present in Application > Storage > Indexed DB ?

**Comment by amitsingh-007** [comment]
Thanks for the reply, I was able to see where I was going wrong. I wasn't using `auth.onAuthStateChanged()` and instead as in the code snippet above, I was initializing the `auth` only when the user signing-in inside a function call. So on opening the popup, there was no code to initialize the auth and I was directly calling the firebase DB update.

For now, I have changed it to such that the `auth` is initialized on loading the app by moving the below code outside any function call. In my case, this works like a charm.
```js
const auth = initializeAuth(firebaseApp, {
  persistence: indexedDBLocalPersistence,
});
```
